added multi step approval docs and new expiry docs (#9)
common-fate/docs: src/content/docs/config/workflows.mdx

  access_duration_seconds  = 60 * 60
  priority                 = 100
+ activation_expiry        = 60 * 60 * 8
+ RequestedToApprovedExpiry = 60 * 60 * 4

> + RequestedToActivateExpiry = 60 * 60 * 4

Reviewer: ```suggestion
+ requested_to_approved_expiry = 60 * 60 * 4
+ requested_to_activate_expiry = 60 * 60 * 4
```

}
```

> The `activation_expiry` cause requests to expire if they are approved but not *activated* for a specified amount of time following being approved.

Reviewer 2: ```suggestion
The `activation_expiry` causes requests to expire if they are approved but not *activated* for a specified amount of time following being approved.
```

> The `RequestedToApprovedExpiry` cause requests to expire if they are requested but not *approved* for a specified amount of time following being requested.

Reviewer 2: ```suggestion
The `RequestedToApprovedExpiry` causes requests to expire if they are requested but not *approved* for a specified amount of time following being requested.
```

Reviewer: ```suggestion
The `requested_to_approved_expiry` cause requests to expire if they are requested but not *approved* for a specified amount of time following being requested.
```

> The `RequestedToApprovedExpiry` causes requests to expire if they are requested but not *approved* for a specified amount of time following being requested.

Reviewer: ```suggestion
The `requested_to_activate_expiry` causes requests to expire if they are requested but not *approved* for a specified amount of time following being requested.
```

> The `RequestedToActivateExpiry` cause requests to expire if they are approved but not *activated* for a specified amount of time following being requested.

Reviewer 2: ```suggestion
The `RequestedToActivateExpiry` causes requests to expire if they are approved but not *activated* for a specified amount of time following being requested.
```

> The `RequestedToActivateExpiry` causes requests to expire if they are approved but not *activated* for a specified amount of time following being requested.

Reviewer: ```suggestion
The `requested_to_activate_expiry` causes requests to expire if they are approved but not *activated* for a specified amount of time following being requested.
```

This approach ensures that extensions are used thoughtfully and only when necessary.

## Muli-step Approvals
Common Fate v2.8.0 introduced the ability to specify multi step approval flows inside Access Workflows.
All conditions / steps on the approval flow must pass before the end user can activate a request.

Multi-step approvals follow similar syntax to cedar policies in that they have a `when` clause to evaludate the reviewing user.

```
resource "commonfate_access_workflow" "workflow-demo" {
    name                     = "demo"
    access_duration_seconds  = 60 * 60
    priority                 = 100
    approval_steps = [
        {
        name = "Approval from Jack"
        when = <<EOF
        principal == CF::User::"usr_2ixfjkldsjlk23z09b1Da"
        EOF
        },
        {
        name = "Apporoval is Required from one of the Security Team"
        when = <<EOF
        principal in Entra::Group::"Security"
        EOF
        },
    ]
  }
```

Results in the following approval steps on a request for access.

![Multi step approval.](assets/multi-step.png)

Multi-step approvals are designed to require each step to map to a different review condition.

> This means that even if 1 user matches more than one of the approval criteria they will only satisfy one of the conditions.The other step will require action from a different user.

Reviewer 2: ```suggestion
This means that even if a user matches more than one of the approval criteria they will only satisfy one of the conditions.The other step will require action from a different user.
```

> Multi-step approvals are designed to require each step to map to a different review condition.
> This means that even if a user matches more than one of the approval criteria they will only satisfy one of the conditions.The other step will require action from a different user.

Reviewer: ```suggestion
Multi-step approvals require each step to be completed by a different reviewer.
When a reviewer matches more than one of the approval steps, they will be automatically assigned to one. The other step will require action from a different user.

In cases where multiple reviews are required and the reviewers match more than one of the steps, Common Fate will automatically reassign the reviewers to the steps such that all steps are satisfied and the request is approved.
```

4. With a maximum of two extensions, you can add up to 20 minutes to your initial access duration.

This approach ensures that extensions are used thoughtfully and only when necessary.

> ## Muli-step Approvals

Reviewer: ```suggestion
## Multi-step Approvals
```

Common Fate v2.8.0 introduced the ability to specify multi step approval flows inside Access Workflows.
All conditions / steps on the approval flow must pass before the end user can activate a request.

> Multi-step approvals follow similar syntax to cedar policies in that they have a `when` clause to evaludate the reviewing user.

Reviewer: ```suggestion
Multi-step approvals follow similar syntax to cedar policies in that they have a `when` clause to evaluate the reviewing user.

The when expression can reference the `principal` and/or the `resource.grant`.
```
